src/model/codacyInput.ts

```diff
@@ -12,7 +12,7 @@ export class Parameter {
 
 export class Pattern {
   readonly patternId: string
-  readonly parameters: Parameter[]
+  parameters: Parameter[]
```

Comment:
Why don't you generate a new pattern maintaining immutability instead mutating it in place?

Reply:
The idea was to mutate our Codacyrc so the tool already has that parameter. This way I am explicitly saying that we are changing this. Otherwise shouldn't be better just to send the Codacyrc and the spec separated to the tool?

I guess I can pass a new immutable codacyrc, but I feel bit like lying. But we do gain the immutability good stuff and less error prone, so it might be worth it 👍